vspace/src/main/java/edu/asu/diging/vspace/core/services/impl/ContentBlockManager.java

```diff
@@ -120,9 +123,28 @@ public CreationReturnValue createImageBlock(String slideId, byte[] image, String
         IImageBlock imgBlock = imageBlockFactory.createImageBlock(slide, slideContentImage);
         imgBlock.setContentOrder(contentOrder);
         ImageBlock imageBlock = imageBlockRepo.save((ImageBlock) imgBlock);
-
         returnValue.setElement(imageBlock);
         return returnValue;
     }
+    
+    @Override
+    public void deleteTextBlockById(String id) throws BlockDoesNotExistException {
+        try {
+            textBlockRepo.deleteById(id);
+        } catch (IllegalArgumentException | EmptyResultDataAccessException exception) {
+            throw new BlockDoesNotExistException(exception);
+        }
+
+    }
+    
+    @Override
+    public void deleteImageBlockById(String id) throws BlockDoesNotExistException {
+        try {
+           imageBlockRepo.deleteById(id);
+        } catch (IllegalArgumentException | EmptyResultDataAccessException exception) {
+            throw new BlockDoesNotExistException(exception);
+        }
```

Comment:
is deleteById actually throwing 'EmptyResultDataAccessException'? And IllegalArgumentException is a runtime exception that is thrown when id is null I believe, as such it does not need to be caught if it can be avoided, which it can by checking if id is null first.